plugins/modules/supervisorctl.py

```diff
@@ -230,6 +230,8 @@ def take_action_on_processes(processes, status_filter, action, expected_result):
         if len(processes) == 0:
             module.exit_json(changed=False, name=name, state=state)
 
+        take_action_on_processes(processes, lambda s: s in ('RUNNING', 'STARTING'), 'stop', 'stopped')
```

Comment:
Please note that `take_action_on_processes` calls `module.exit_json()`, which means that the next lines will no longer be executed. You need to adjust that function so you can control whether it should do that (should be the default) or not (needed here).

Reply:
You are right, I didn't tested it thoroughly.